modules/features2d/test/test_descriptors_regression.impl.hpp

```diff
@@ -145,6 +173,25 @@ class CV_DescriptorExtractorTest : public cvtest::BaseTest
             ts->set_failed_test_info( cvtest::TS::FAIL_INVALID_TEST_DATA );
         }
 
+        image = prepare_img(false);
+        Mat dbl;
+        try
+        {
+            double_image(image, dbl);
+
+            Mat downsized_back(dbl.rows/2, dbl.cols/2, CV_32F);
+            resize(dbl, downsized_back, Size(dbl.cols/2, dbl.rows/2), 0, 0, INTER_NEAREST);
+
+            cv::Mat diff = (image != downsized_back);
+            ASSERT_TRUE(cv::countNonZero(diff) == 0);
```

Comment:
> countNonZero

Avoid using of `countNonZero()`.
Use `ASSERT_LE(cv::norm(image, downsized_back), 1e-5);` instead.

Reply:
@alalek, nearest neighbor resize is used here. It should be bit exact. So the suggested method to compare image and downsized_back image is correct

Reply:
For bit-exact compare with 0:

```
ASSERT_EQ(0, cv::norm(image, downsized_back, NORM_INF));
```